cpmpy/expressions/core.py

```diff
@@ -329,10 +329,11 @@ def __rmod__(self, other):
 
     def __pow__(self, other, modulo=None):
         assert (modulo is None), "Power operator: modulo not supported"
-        if other == 0:
-            return 1
-        elif other == 1:
-            return self
+        if is_num(other):
```

Comment:
Why?

Reply:
The test `if other == 0` will always go through when `other` is an expression (will return a new expression). Alternatively we can do `if other == 0 is True` but I think this is more clear

Reply:
oh wauw that's a pretty serieus bug actually :o